backends/mediatek/runtime/include/NeuronBufferAllocator.h

```diff
@@ -102,7 +107,7 @@ class MemoryUnit {
   size_t mSize = 0;
 };
 
-class BufferAllocator : public MemoryAllocator {
+class BufferAllocator : public executorch::runtime::MemoryAllocator {
```

Comment:
It looks like there is prebuilt code that refers to ET (executorch) types: specifically, this subclass of MemoryAllocator.

ET does not guarantee ABI compatibility across versions, or even between git commits. If the definition of the MemoryAllocator base class were to change in an incompatible way (new field, different vtable order), then BufferAllocator would misbehave when interpreted as a `MemoryAllocator*`: at best, crashes; at worst, subtle memory corruption or execution of random code.

It is not safe for prebuilt code to refer to ET types.

Reply:
Thank you for bringing this to our attention. We are currently conducting an internal review of the library's source code, which will be made publicly available upon completion. This will eliminate the need for prebuilt binaries.
In the meantime, we will establish CI to ensure MTK backend works with the OSS models.

Reply:
I created https://github.com/pytorch/executorch/issues/6074 to track this issue. Please comment on that task if you have any questions for me!